### DISCUSSION TITLE
Filament runout sensor when printing via OctoPrint

### DISCUSSION THREAD

**Discussion by Schnabulation** [Jyers/Marlin#148] [Q&A] [answered]
Quick question: When printing from OctoPrint (i.e. Host Print) will the filament runout sensor work and pause the print?

I have read a couple of threads already that say OctoPrint will not know when the sensor is triggered and continue to stream G-code to the printer.

**Comment by t3chguy** [comment]
> I have read a couple of threads already that say OctoPrint will not know when the sensor is triggered and continue to stream G-code to the printer.

Sure but there are mechanisms for signalling the buffer is full. How is it different than running a long blocking GCODE like M600?

**Reply by Schnabulation** [reply]
I don't know, that's why I ask. My runout sensor should arrive this week and I wanted to clearify in advance.

Do you happen to know if it correctly triggers the runout when the sensor is attached to the dedicated port on the mainboard and printing from OctoPrint?

**Reply by t3chguy** [reply]
In my case, where runout in Marlin is configured to execute M600 it works just fine. I don't think it tells OctoPrint it is Paused so it may mess up progress estimates there

**Reply by strayr** [reply]
For octoprint to work properly you need 
HOST_ACTION_COMMANDS and HOST_PROMPT_SUPPORT if you want octoprint to know when you paused/stooped a print and to be able to answer prompts from octoprint respectively

Advanced OK also helps here.

**Reply by jjfs127** [reply]
> I don't know, that's why I ask. My runout sensor should arrive this week and I wanted to clearify in advance.
> 
> Do you happen to know if it correctly triggers the runout when the sensor is attached to the dedicated port on the mainboard and printing from OctoPrint?

Out of curiosity, what runout sensor have you ordered?

**Reply by Schnabulation** [reply]
I have bought the original Creality runout sensor kit for the E3 V2. It comes pre-mounted on a plate that replaced the extruder motor mounting plate.

**Comment by Jyers** [comment, marked as answer]
Just gonna conclude this here for anyone who needs this. As @strayr said, if you enable HOST_ACTION_COMMANDS and HOST_PROMPT_SUPPORT, octoprint will receive a command when filament runout occurs and pause the print.

**Reply by Schnabulation** [reply]
Thank you, yes, I can confirm this: filament runout works perfectly fine when printing from OctoPrint.

**Reply by HawkMan79** [reply]
> Just gonna conclude this here for anyone who needs this. As @strayr said, if you enable HOST_ACTION_COMMANDS and HOST_PROMPT_SUPPORT, octoprint will receive a command when filament runout occurs and pause the print.

Are these enabled in the pre compiled releases ?

**Reply by Jyers** [reply]
Not currently, but they will be in the next release

**Reply by strayr** [reply]
This is probably a good choice, maybe with default/example config files annotated with advice to disable to save memory IFF user/distributor doesn't use octoprint or similar.

**Reply by eecue** [reply]
Following up here, are you still planning on including these in the next release @Jyers

**Reply by Jyers** [reply]
@eecue New releases have already come out since then, it should now be enabled in the precompiled versions.

**Reply by eecue** [reply]
@Jyers hmm, mine just stopped printing on the printer and then Octoprint said something about serial failed after sending data to the printer.  Running most recent version of firmware and Octoprint.  Do I need to configure something on the screen/GCode/Octoprint?

**Reply by Jyers** [reply]
Interesting, do you have your M600 load and unload distances set in Control>Advanced? That is the only thing I could imagine causing the pause to fail.

**Reply by eecue** [reply]
Hmm I actually don’t. I wasn’t sure what to set those to. Is that how far it needs to feed to back out from the hot end until it’s no longer in the extruded gears?

**Reply by Jyers** [reply]
Yes, it should be approximately the distance from the hotend to the gears

**Reply by strayr** [reply]
You can set the basic load/unload length with m603, set the unload big and measure how much filament is kicked out, and use that to set the loading length. 

A fairly long slow purge until cancelled behaviour is handy and lets you get away with a very fast load with long Bowden tube, but that needs the distance setting correctly and a bit short. Fixing snapped filament or putting a new roll on doesn't need much of a purge, going from vividly coloured filament to white or natural requires more purging. Leave yourself some flexibility there. 

A slower initial feed is less likely to do something really bad getting the filament end back through the extruder.

I've been using up roll ends for the last few weeks and have got to know m600 well.

**Reply by eecue** [reply]
Got it and makes sense thanks!
